File: 033_search_in_rotated_sorted_array.java
class Solution {
    public int search(int[] nums, int target) {
        int offset = 0;
        for (int i = 1; i < nums.length; i++){
            if (nums[i] < nums[offset]){
                offset = i;
                break;
            }
        }

        Arrays.sort(nums);
        int low = 0, high = nums.length - 1, mid;
        while(low <= high){
            mid = (low + high) / 2;
            if (target < nums[mid]){
                high = mid - 1;
            }else if (target > nums[mid]){
                low = mid + 1;
            }else{
                int index = mid - (nums.length - offset);
                if (index < 0) index = nums.length + index;
                return index;
            }
        }
        return -1;
    }
}

File: python/033_search_in_rotated_sorted_array.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        offset = 0
        for i in range(1, len(nums)):
            if nums[i] < nums[offset]:
                offset = i
                break

        nums.sort()
        lo, hi = 0, len(nums) - 1
        while lo <= hi:
            mid = (lo + hi) // 2
            if nums[mid] > target:
                hi = mid - 1
            elif nums[mid] < target:
                lo = mid + 1
            else:
                index = mid - (len(nums) - offset)
                if index < 0:
                    index = len(nums) + index
                return index
        return -1


